### DISCUSSION TITLE
Replacement for codeunit EffectivePermissionMgt?

### DISCUSSION THREAD

**Discussion by quadfin** [microsoft/ALAppExtensions#18012] [Q&A] [answered]
Since the Permission table is marked for removal and codeunit 9852 "Effective Permissions Mgt." depends on it, is there an equivalent implementation not dependent on the Permission table?

I.e,, without referencing the Permission table:
- How should one check for granular permissions (RIMDX) at the object level?
- How to test if a specific user has permissions (one of RIMDX) on a given object?

**Comment by JesperSchulz** [comment, marked as answer]
See related [Yammer thread on the topic](https://www.yammer.com/dynamicsnavdev/#/Threads/show?threadId=1714126488625152). The thread hasn't landed on a conclusion yet, but I'll look into it once more and update that thread.

**Reply by JesperSchulz** [reply]
The Yammer thread concludes that the "Expanded Permission" table is the answer to your question :-)
